core/domain/skill_services.py

```diff
@@ -1065,6 +1065,36 @@ def skill_has_associated_questions(skill_id):
     return len(question_ids) > 0
 
 
+def filtered_skills_with_none_type_keys(degrees_of_mastery):
+    """Given a dict of skills id's with there corresponding
+    mastery. It first split the dict and then returns a list of skill
+    id.
+
+    Args:
+        degrees_of_mastery: dict. The dict of type (skill_ids, float|None)
+
+    Returns:
+        list. Sorted list of skill ids.
+    """
+
+    skill_dict_with_float_value = {
+        skill_id: degree for skill_id, degree in degrees_of_mastery.items()
+        if degree is not None}
+
+    sorted_skill_ids_with_float_value = sorted(
+        skill_dict_with_float_value, key=skill_dict_with_float_value.get)
+
+    skill_ids_with_none_value = [
+        skill_id for skill_id, degree in degrees_of_mastery.items()
+        if degree is None]
+
+    if feconf.MAX_NUMBER_OF_SKILL_IDS <= len(skill_ids_with_none_value):
+        return skill_ids_with_none_value[:feconf.MAX_NUMBER_OF_SKILL_IDS]
+    else:
+        return (skill_ids_with_none_value + sorted_skill_ids_with_float_value[:(
+            feconf.MAX_NUMBER_OF_SKILL_IDS - len(skill_ids_with_none_value))])
```

Comment:
Can we do something like:
```suggestion
	sorted_skill_ids = (
	    skill_ids_with_none_value + sorted_skill_ids_with_float_value)
    return sorted_skill_ids[:feconf.MAX_NUMBER_OF_SKILL_IDS]
```

Reply:
Sure, I was thinking maybe the way I wrote can be a bit faster.. but I guess it's making code look more unreadable, will update it. Thanks!